lobby/build.gradle

```diff
@@ -12,6 +12,8 @@ dependencies {
     implementation project(':java-extras')
     implementation project(':game-core')
     implementation project(':lobby-db-dao')
+
+    implementation 'javax.mail:mail:1.5.0-b01'
```

Comment:
I'm not sure about this, but I believe there's a newer, maintained version of this:
https://search.maven.org/artifact/jakarta.mail/jakarta.mail-api/1.6.3/jar
Looks like this library was taken over by the eclipse foundation. Telling by the GitHub repository it looks very similar: https://github.com/eclipse-ee4j/mail

game-core also seems to depend on a similar version of the API: `com.sun.mail:javax.mail:1.6.2` unifying all the versions would be desireable.

Reply:
That is `mail-api`; I recall reading something that if you want to actually send mail, you need the implementation dependency. Given the testing done, I'm not sure if there is any concrete benefit to going for a different version.

Reply:
From what I can tell the newer version doesn't throw UnsupportedEncodingException

Reply:
Removing exactly 2 lines for the catch in a small class strikes me as a very marginal benefit.

Is there a non 'mail-api' version? Is this worth the research and effort we are putting into it?

Reply:
(And worth the re-testing)

Reply:
Nevermind doesn't seem to actually be the case :(

Reply:
> com.sun.mail:javax.mail:1.6.2 unifying all the versions would be desireable.

I would agree to reducing dependencies and keeping them the same. I'll re-test with the com.sun dependency. I'm a bit surprised though we can / should use a `com.sun` given the license we use.

Reply:
Actually...  Are we *sure* that `com.sun.mail` is a cool dependency to have??

I could see us getting away with that as a server side dependency, but if we distribute that with the game-core, it could become a bigger deal.

Reply:
Looks like the com.sun.mail library is GPLv2, should be okay then. I still have my test program available, can verify in 5 minutes...

Reply:
Well I think its desireable to use a library that is well-maintained, even if we don't get any large benefits from it.
Not necessarily as part of this PR, but as a follow-up PR I think it would make sense to bump the versions to the latest jakarta version. IMO it's not that unlikely that this would work right out of the box because it doesn't look like the API changed at all.

Reply:
Updated package, re-testing looked good: 3227d4781a83f005a612621ddaccd3f922b2eec9

Reply:
Nice 👍 

If you have the time, consider checking if https://github.com/eclipse-ee4j/mail/releases compiles, just to be sure this is actually the same library with a new owner.
If it compiles might be worth updating at some point in the future